### DISCUSSION TITLE
[archived] "Configure Artifact Repository" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#10320] [General]
# en/latest/configure-artifact-repository/

https://argo-workflows.readthedocs.io/en/latest/configure-artifact-repository/

**Comment by Is4b3lla3** [comment]
The minio configuration is missing the config section：
```yaml
# ...
data:
  config: |
    artifactRepository:
      s3:
        bucket: my-bucket  # 默认的 bucket 名称
```

**Reply by haiouc** [reply]
https://helm.min.io , It is no longer updated. I haven't seen any documentation on the argo workflow of the artifact (mini operator)

**Reply by agilgur5** [reply]
You can use https://charts.min.io/ instead. It's being updated in the docs in #12438

**Comment by wichert** [comment]
Contrary to what this page says, the quickstart templates (at least for v3.4.4) do not include minio.

**Reply by caelan-io** [reply]
Thanks for reporting! PR [open](https://github.com/argoproj/argo-workflows/pull/10448) to update the doc.

**Comment by pinkgizmo** [comment]
Hi, I have deployed argo-workflow controller and config in a namespace "argo" but my workflows are deployed in specific project namespaces.)
Argo is configured to watch all my cluster's namespaces and it works very well.

Question is : can I configure the artifactRepository config only in my main "argo" namespace and use it in workflows deployed in my other namespaces ?
For now I have a configMap in each namespace but I'd like to deployed it once in my main "argo" namespace but be used by every workflow on my cluster.

(I deployed through the official helm chart)

**Reply by pinkgizmo** [reply]
I found out by testing that the artifactRepository config or the equivalent in a dedicated configmap is read by workflows on others namespace. 
Unfortunately, as this config need to store the access key in a secret, the workflow try to read a secret from the project namespace and not from the main namespace (I use azure storage in my context).
So It can't totally be a shared config, each project have to implement at least this secret to access the storage.

**Comment by captaincoordinates** [comment]
Following the instructions exactly, external-IP is always "pending" and `localhost:9000` is not accessible:
```
> kubectl get service argo-artifacts

NAME             TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
argo-artifacts   LoadBalancer   10.110.1.175   <pending>     9000:30849/TCP   6m28s
```

**Reply by Paritosh-Anand** [reply]
That might be cause of no ingress controller. 
You can still access the minio ui, for the dev/local setup

Minikube

```
minikube service --url argo-artifacts
```

or use NodePort

**Reply by captaincoordinates** [reply]
I've seen this behaviour inconsistently after running `helm install argo-artifacts minio/minio --set service.type=LoadBalancer --set fullnameOverride=argo-artifacts` on Kubernetes within Docker Desktop. From what I've seen it may be related to how long the Kubernetes cluster has been up. Executing the command without waiting a few minutes can create this problem.

**Comment by everestas** [comment]
If I set up default artifact by changing workflow controller configmap according to the example, how does Argo know what endpoint should be used? I imagine this needs to specified in the configmap as well, right?

**Reply by byrisettya** [reply]
@everestas When a workflow that uses artifact repository is created, workflow controller will look for necessary configuration of artifactory in the below order

1. artifactRepositoryRef field in the workflow spec
2. configmap with the name artifact-repositories in the workflow deployment namespace
3. workflow controller configmap

When, the configuration of artifact repositories in not found in the first two places above, that's when it looks at workflow controller configmap.

**Reply by Hanliber** [reply]
when I configure the default artifact repository: How do I set the endpoint in workflow-controller-configmap? 
My default repository is MinIO，and My MinIO external IP is http://192.168.49.2:32377，so the my endpoint is http://192.168.49.2:32377?

**Comment by shenshanyoumu** [comment]
in argo workflow within same pod I has define several volumeMountPath, is possible only upload specific volumeMountPath to azure Storage?

**Reply by agilgur5** [reply]
Artifacts require you to specify a path, so by definition, yes.

**Comment by haiouc** [comment]
https://helm.min.io , It is no longer updated. I haven't seen any documentation on the argo workflow of the artifact (mini operator)

**Reply by agilgur5** [reply]
You can use https://charts.min.io/ instead. It's being updated in the docs in #12438

**Comment by beck3905** [comment]
It appears the min.io installation described by these docs was deprecated 2 years ago and as of about 2 weeks ago those helm charts are unavailable. Are there any updated docs using the new min.io helm charts? See https://github.com/minio/minio/discussions/13759#discussioncomment-7710330 for more info.

I'm completely new to Argo and Min.io. I'm trying to create a new workflow that uses artifacts and it was at first unclear from the documentation that I have to configure an artifact repository and there is not a default one installed with Argo Workflows. Now I am struggling to figure out how to install Min.io from their new operator/tenant helm charts and get those working with Argo Workflows.

**Reply by agilgur5** [reply]
You can use https://charts.min.io/ instead. It's being updated in the docs in #12438.

https://github.com/minio/minio/discussions/13759#discussioncomment-7721214 mentions https://github.com/minio/minio/tree/master/helm/minio which  uses https://charts.min.io/

**Comment by manju-rn** [comment]
How to use the `HTTP` as a Artifact repository?  Does the endpoint has to be like an Webserver or FTP server?

**Reply by agilgur5** [reply]
It is a bit under documented right now. It is similar to Artifactory, see the code that supports both [here](https://github.com/argoproj/argo-workflows/blob/88304568af5b8222344cde8e5b96d1a8484a50f8/workflow/artifacts/http/http.go).

There's one example of `inputs` in the [Hardwired Artifacts page](https://argo-workflows.readthedocs.io/en/latest/walk-through/hardwired-artifacts/) which is a `GET` request. `outputs` are a `PUT` request.

It would not be FTP as FTP is a different protocol from HTTP.

**Comment by carvarr** [comment]
I've configured azure artifact rep using following setting

```yaml
apiVersion: v1
data:
  config: |
    artifactRepository:
      azure:
        container: argo-workflows
        endpoint: https://xxxx.blob.core.windows.net
        useSDKCreds: true
```

My aks cluster uses a managed identity in order to autenticate against my storage 

When I'm trying to run a workflow I'm getting this error: 
```
unable to upload file /tmp/argo/outputs/artifacts/hello-art.tgz to Azure: no Host in request URL
```

Anybody can help me overcome this ? :/

**Comment by patialashahi31** [comment]
Can we use non default artifacts in a cloud-agnostic way? Lets say I am not sure what kind of artifact repo I am using and in case of blob I will need to add container and in case of s3 I need to add bucket?

**Reply by agilgur5** [reply]
You might be able to do this with [Conditional Artifacts](https://argo-workflows.readthedocs.io/en/latest/conditional-artifacts-parameters/) or some other indirection, e.g. a `template` that runs other `templates` depending on your condition
